addons/account_voucher/migrations/9.0.1.0/post-migration.py

```diff
@@ -71,9 +71,21 @@ def create_payments_from_vouchers(env):
         SET payment_id = av.id
         FROM account_voucher_line avl
         JOIN account_voucher av ON av.id = avl.voucher_id
-        WHERE avl.move_line_id=aml.id
+        WHERE avl.move_line_id = aml.id
         AND av.voucher_type IN ('receipt', 'payment')
-        AND av.state in ('draft', 'posted')
+        AND av.state IN ('draft', 'posted')
+        AND avl.reconcile
```

Comment:
@NL66278 It is not correct to put this line, because in odoo 8 it meant the voucher lines whose corresponding account move line is being fully paid. Sometimes you pay just part of the move line in the voucherm and that will leave avl.reconcile = False, but it is still a payment!

Instead, add the following:
```
AND (aml.id in (select credit_move_id from account_partial_reconcile) 
OR aml.id in (select debit_move_id from account_partial_reconcile))

```